### PR TITLE
feat(argo-cd): Add argocd image updater

### LINKED ISSUE #826: image updater chart
**Describe the solution you'd like**
I like to use a helm chart for image updater project

**Describe alternatives you've considered**
installing the component using terraform its hard to maintain

### PR DESCRIPTION
**Pull request by hobbsh**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [ ] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

This resolves #826

### CODE REVIEW

**Comment by juv** [comment]
I'm looking forward to this! 

@hobbsh do you know how the argocd-helm reviewers take notice of this pr? just to speed up the merge process a little bit

**Comment by hobbsh** [comment]
> @hobbsh do you know how the argocd-helm reviewers take notice of this pr? just to speed up the merge process a little bit

I am not sure who would be responsible for getting this merged in. @seanson, who do I need to loop in to get this pushed through?

**Review by somaliz — APPROVED** [review]

**Comment by hobbsh** [comment]
@alexec Sorry if you're not the right person but I'm wondering who can help get this pushed through? Thanks!

**Comment by mkilchhofer** [comment]
I wonder, why this PR got closed. In my opinion it would make sense to reopen this and update it to the latest manifests in the release 0.9.4 (https://github.com/argoproj-labs/argocd-image-updater/releases)

File: charts/argocd-image-updater/templates/rbac.yaml
Comment (oliverbaehler) [inline charts/argocd-image-updater/templates/rbac.yaml:30]:
```suggestion
{{- if .Values.rbac.enabled }}
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  labels:
    {{ include "argocd-image-updater.labels" . | nindent 4 }}
  name: {{ include "argocd-image-updater.fullname" . }} 
rules:
- apiGroups:
  - ""
  resources:
  - secrets
  verbs:
  - get
  - list
  - watch
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  labels:
    {{ include "argocd-image-updater.labels" . | nindent 4 }}
  name: {{ include "argocd-image-updater.fullname" . }}
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: {{ include "argocd-image-updater.fullname" . }}
subjects:
- kind: ServiceAccount
  name: {{ include "argocd-image-updater.serviceAccountName" . }}
{{- end }}  
```

File: charts/argocd-image-updater/values.yaml
Comment (oliverbaehler) [inline charts/argocd-image-updater/values.yaml:23]:
```suggestion

# Enable RBAC creation
rbac:
  enabled: true


```

**Review by oliverbaehler — CHANGES_REQUESTED** [review]

**Comment by hobbsh** [comment]
Thanks for the reviews! I'll try to take a pass at everything later today.

File: charts/argocd-image-updater/Chart.yaml
Comment (hobbsh) [reply charts/argocd-image-updater/Chart.yaml:16]: Done, thanks

File: charts/argocd-image-updater/values.yaml
Comment (hobbsh) [reply charts/argocd-image-updater/values.yaml:23]: All set 👍

File: charts/argocd-image-updater/templates/rbac.yaml
Comment (hobbsh) [reply charts/argocd-image-updater/templates/rbac.yaml]: Updated the Role rules and put everything behind the `rbac.enabled` value

File: charts/argocd-image-updater/templates/configmap.yaml
Comment (hobbsh) [reply charts/argocd-image-updater/templates/configmap.yaml]: I prefer to leave the `toYaml` on the next line for readability (if that's ok) but fixed the chompstyle. Thanks!

**Comment by travis-sobeck** [comment]
I would really like to see this be a thing ... I see it was closed, then re-opened ... ? Is this moving forward?  Thanks.

**Comment by hobbsh** [comment]
> I would really like to see this be a thing ... I see it was closed, then re-opened ... ? Is this moving forward? Thanks.

I think it's ready to merge, just waiting for a re-review from @mkilchhofer and @oliverbaehler

File: charts/argocd-image-updater/Chart.yaml
Comment (mkilchhofer) [inline charts/argocd-image-updater/Chart.yaml:8]:
```suggestion
icon: https://argocd-image-updater.readthedocs.io/en/stable/assets/logo.png
```

File: charts/argocd-image-updater/Chart.yaml
Comment (mkilchhofer) [inline charts/argocd-image-updater/Chart.yaml:16]:
```suggestion
  - name: argoproj
    url: https://argoproj.github.io/
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
A chart README.md would be nice to have. Additional I'd use the github organization as a maintainer as the mentioned 3 humans are maintainers of the software, not the chart itself :-)

**Comment by hobbsh** [comment]
> A chart README.md would be nice to have. Additional I'd use the github organization as a maintainer as the mentioned 3 humans are maintainers of the software, not the chart itself :-)

Sure, I'll work on a readme tonight. 👍

**Comment by mkilchhofer** [comment]
> Sure, I'll work on a readme tonight. 👍

Perfect :-). If you want you could have a look into helm-docs (#696). I already used it in another project:
- helm values comments: https://github.com/gogatekeeper/helm-gogatekeeper/blob/main/charts/gatekeeper/README.md.gotmpl
- README.md.gotmpl: https://github.com/gogatekeeper/helm-gogatekeeper/blob/main/charts/gatekeeper/README.md.gotmpl
- Then run `helm-docs` which should generate the final README.md

(commit both README.md and README.md.gotmpl) :)

**Comment by hobbsh** [comment]
> > Sure, I'll work on a readme tonight. 👍
> 
> Perfect :-). If you want you could have a look into helm-docs (#696). I already used it in another project:
> 
> * helm values comments:
>   https://github.com/gogatekeeper/helm-gogatekeeper/blob/main/charts/gatekeeper/values.yaml#L8-L16
> * README.md.gotmpl:
>   https://github.com/gogatekeeper/helm-gogatekeeper/blob/main/charts/gatekeeper/README.md.gotmpl
> * Then run `helm-docs` which should generate the final README.md
> 
> (commit both README.md and README.md.gotmpl) :)

Yep, `helm-docs` is great. I was going to do that in conjunction with a few specifics on some values.

**Review by mkilchhofer — APPROVED** [review]

**Comment by mkilchhofer** [comment]
@oliverbaehler everything ok from your side? :-) I would merge then.

**Review by oliverbaehler — APPROVED** [review]
Lgtm

**Comment by mkilchhofer** [comment]
Tadaaaaa 🥳 🎺

**Pull request merged**
